### PR TITLE
AudioCommon: release mutex explicitly to avoid -Wunused-result warning

### PR DESCRIPTION
**Pull request by Tilka**

### CODE REVIEW

**Review by sepalani — COMMENTED** [review]
The code looks good to me.

I'm surprised that we're not using atomic wait/notify instead of using `conditional_variable`+`mutex`.

I don't think we're subject to this [cv race](https://en.cppreference.com/w/cpp/thread/condition_variable):

> Even if the shared variable is atomic, it must be modified while owning the mutex to [correctly](https://stackoverflow.com/questions/38147825/) publish the modification to the waiting thread. 

The current ALSA code doesn't look obvious but since we're locking before notifying the `cv`, the `cv` shouldn't miss the change as [explained here](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/Event.h#L33-L42). So this stackoverflow example doesn't seem to apply here.

File: Source/Core/AudioCommon/AlsaSoundStream.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ AlsaSound::~AlsaSound()
   m_thread_status.store(ALSAThreadStatus::STOPPING);
 
   // Immediately lock and unlock mutex to prevent cv race.
-  std::unique_lock<std::mutex>{cv_m};
+  std::unique_lock<std::mutex>{cv_m}.release();
```
Comment (AdmiralCurtiss) [inline Source/Core/AudioCommon/AlsaSoundStream.cpp:23]: Err, is that right? https://en.cppreference.com/w/cpp/thread/unique_lock/release doesn't unlock the mutex.

Comment (sepalani) [reply Source/Core/AudioCommon/AlsaSoundStream.cpp:23]: My bad, I think you're correct, I thought `release()` also unlock the mutex. So I guess `unlock()` should be call instead.

Comment (Tilka) [reply Source/Core/AudioCommon/AlsaSoundStream.cpp]: Thanks for catching that!

**Pull request merged**